File: api/src/main/java/com/nidus/reserva/application/port/input/ReservaService.java
package com.nidus.reserva.application.port.input;

import com.nidus.reserva.application.dto.CrearReservaRequest;
import com.nidus.reserva.application.dto.ModificarReservaRequest;
import com.nidus.reserva.application.dto.ReservaResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReservaService {

    ReservaResponse crear(CrearReservaRequest request, Long usuarioId);

    ReservaResponse modificar(Long id, ModificarReservaRequest request, Long usuarioId);

    void cancelar(Long id, Long usuarioId);

    ReservaResponse obtenerPorId(Long id, Long usuarioId);

    Page<ReservaResponse> listarPorUsuario(Long usuarioId, Pageable pageable);

    Page<ReservaResponse> listarTodas(Pageable pageable);
}

File: frontend/src/app/core/services/reserva.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { CrearReservaRequest, ModificarReservaRequest, ReservaResponse } from '../models/reserva.models';

@Injectable({ providedIn: 'root' })
export class ReservaService {
  private api = `${environment.apiUrl}/reservas`;

  constructor(private http: HttpClient) {}

  crear(peticion: CrearReservaRequest): Observable<ReservaResponse> {
    return this.http.post<ReservaResponse>(this.api, peticion);
  }

  listarMisReservas(page = 0, size = 20): Observable<any> {
    return this.http.get<any>(`${this.api}?page=${page}&size=${size}`);
  }

  listarTodas(page = 0, size = 20): Observable<any> {
    return this.http.get<any>(`${this.api}/todas?page=${page}&size=${size}`);
  }

  obtener(id: number): Observable<ReservaResponse> {
    return this.http.get<ReservaResponse>(`${this.api}/${id}`);
  }

  modificar(id: number, peticion: ModificarReservaRequest): Observable<ReservaResponse> {
    return this.http.put<ReservaResponse>(`${this.api}/${id}`, peticion);
  }

  cancelar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.api}/${id}`);
  }
}
